tools/check-changelog/check-changelog.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-changelog"
+)
+
+var (
+	skipLabelName = "skip-changelog-check"
+	skipTitles    = []string{"chore", "test", "doc", "ci"} // Dependabot uses chore.
+)
+
+func main() {
+	var (
+		title      = os.Getenv("PR_TITLE")
+		number     = os.Getenv("PR_NUMBER")
+		jsonLabels = os.Getenv("PR_LABELS")
+	)
+	if title == "" || number == "" || jsonLabels == "" {
+		log.Fatal("Environment variables PR_TITLE, PR_NUMBER and PR_LABELS are required")
+	}
+	var labels []string
+	if err := json.Unmarshal([]byte(jsonLabels), &labels); err != nil {
+		log.Fatalf("PR_LABELS is not a stringified JSON array: %v", err)
+	}
+
+	filePath := fmt.Sprintf(".changelog/%s.txt", number)
+	content, errFile := os.ReadFile(filePath)
+	if errFile == nil { // Always validate changelog file if present, skip logic is not considered in this case
+		entry := changelog.Entry{
+			Body: string(content),
+		}
+		if err := entry.Validate(); err != nil {
+			log.Fatalf("Error validating changelog file: %s, err: %v", filePath, err)
+		}
```

Comment:
[nit] could be a function `validChangelog(string(content))`

Reply:
let me think more about that, the thing is that there are many variables in and out

Reply:
done, thanks